vendor/llm2vec_monGARS/llm2vec/loss/loss_utils.py

```
@@ -0,0 +1,107 @@
+    all_gather with gradient back-propagation
+    """
+
+    @staticmethod
+    def forward(ctx, tensor_list, tensor, group, async_op):
+        torch.distributed.all_gather(
+            tensor_list, tensor, group=group, async_op=async_op
+        )
+        return tuple(tensor_list)
+
+    @staticmethod
+    def backward(ctx, *grad_list):
```

Ruff (0.13.3)

Unused static method argument: `ctx`

(ARG004)